api/src/main/java/org/apache/commons/rdf/api/RDFTermFactory.java

```diff
@@ -17,262 +17,61 @@
  */
 package org.apache.commons.rdf.api;
 
-import java.io.Serializable;
-import java.util.Locale;
-
 /**
- * Factory for creating RDFTerm and Graph instances.
- * <p>
- * It is not specified how an implementation should provide a RDFTermFactory.
- * <p>
- * If an implementation does not support a particular method (e.g. it requires
- * additional parameters or can't create graphs), then it MAY throw
- * {@link UnsupportedOperationException}, as provided by the <code>default</code>
- * implementations in this interface.
+ * Factory for creating RDFTerm instances..
  * <p>
- * If a factory method does not allow or support a provided parameter, e.g.
- * because an IRI is considered invalid, then it SHOULD throw
- * {@link IllegalArgumentException}.
- *
- * @see RDFTerm
- * @see Graph
- * @see Quad
+ * This interface is <strong>deprecated</strong> in favour of
+ * the richer {@link RDF}.
+ * 
+ * @see RDF
  */
+@Deprecated
 public interface RDFTermFactory {
 
-    /**
-     * Create a new blank node.
-     * <p>
-     * The returned blank node MUST NOT be equal to any existing
-     * {@link BlankNode} instances according to {@link BlankNode#equals(Object)}.
-     *
-     * @return A new, unique {@link BlankNode}
-     * @throws UnsupportedOperationException
-     *             If the operation is not supported.
-     */
     default BlankNode createBlankNode() throws UnsupportedOperationException {
```

Comment:
Yes, I removed the notion of `UnsupporetedOperationException` from `RDF` as all our bundled implementations are complete and callers should not be too worried about that. 

The Javadoc now say that any partial `RDF` implementations should be _clearly documented as such_:

>  * A <em>partial RDF implementation</em> 
>  * should be clearly documented as such, and may throw 
>  * {@link UnsupportedOperationException} where applicable, e.g. if it 
>  * does not support creating {@link Dataset}s or {@link Quad}s. 

Is this sufficient? E.g. an `RDF` instance within OWLAPI probably would not know how to make a `Graph` or `Dataset` - but would be a bit more exotic factory than the ones bundled within Commons RDF.